homeassistant/components/flipr/config_flow.py

```diff
@@ -85,11 +85,17 @@ def _show_setup_form(self, errors=None):
 
     async def _authenticate_and_search_flipr(self) -> list[str]:
         """Validate the username and password provided and searches for a flipr id."""
-        client = await self.hass.async_add_executor_job(
-            FliprAPIRestClient, self._username, self._password
-        )
 
-        flipr_ids = await self.hass.async_add_executor_job(client.search_flipr_ids)
+        def create_api_client_and_search_flipr_ids(
```

Comment:
I looked u the library and the client instantiation doesn't seem to have any side effects. So we can create the client in async context and then we don't need this extra function since there's only one sync call.